src/sql/azdata.proposed.d.ts

```diff
@@ -1790,6 +1795,77 @@ declare module 'azdata' {
 		 * The path of the parent node.
 		 */
 		parentNodePath: string;
+		/**
+		 * flag to indicate if the node supports filtering
+		 */
+		isFilteringSupported?: boolean;
+		/**
+		 * list of filters that are supported by the node
+		 */
+		defaultFilters?: NodeInfoFilterDefinition[];
+		/**
+		 * list of properties to filter children nodes on
+		 */
+		filters?: NodeInfoFilter[];
```

Comment:
Why do this need to be on the nodes themselves? You already pass the filters in the expand operation which is where the filtering will happen so what would this be used for?

Reply:
This is because the tree will contain nodes with filters, and we can call standard tree operations like expand/updateChildren on those nodes with filters already present in them.